src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -448,6 +479,20 @@ boost::shared_ptr<ConsoleProcess> ConsoleProcess::fromJson(
    else
       pProc->exitCode_.reset(exitCode.get_int());
 
+   // Newly added in v1.1; do checked access below here
+   //-------------------------------------------------
+   json::Value terminalHandle = obj["terminal_handle"];
```

Comment:
Use `readObject` for this idiom; it will automatically error if the property doesn't exist or is the wrong type. As written this can throw if `obj` does not have the named property (and as you've probably noticed we usually trap exceptions close to the source). https://github.com/rstudio/rstudio/blob/54cd3abcfc58837b433464c793fe9b03a87f0bb4/src/cpp/core/include/core/json/JsonRpc.hpp#L515-L517